diff --git a/lisp/xwidget.el b/lisp/xwidget.el
index cca01c8..c5a84db 100644
--- a/lisp/xwidget.el
+++ b/lisp/xwidget.el
@@ -81,7 +81,7 @@ This returns the result of `make-xwidget'."
 (defun xwidget-at (pos)
   "Return xwidget at POS."
   (let* ((disp (get-text-property pos 'display))
-         (xw (car (cdr (cdr disp)))))
+         (xw (ignore-errors (car (cdr (cdr disp))))))
     (when (xwidget-live-p xw) xw)))
 
 
@@ -378,7 +378,7 @@ one char."
              (> (window-hscroll) 0))
         (set-window-hscroll nil (- (window-hscroll) n))
       (xwidget-webkit-execute-script session
-                                     (format "window.scrollBy(%-d, 0);"
+                                     (format "window.scrollBy(-%d, 0);"
                                              (* n (window-font-width)))))))
 
 (defun xwidget-webkit-scroll-top ()
